sdk/tables/Azure.Data.Tables/src/TableEntity.cs

```diff
@@ -235,5 +236,56 @@ private static void EnforceType(Type requestedType, Type givenType)
                     $"Cannot return {requestedType} type for a {givenType} typed property."));
             }
         }
+
+        /// <summary>
+        /// Performs type coercion for numeric types.
+        /// Integers will be coerced to long or double.
+        /// Double and long values will accept integers values but remain double or long.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">The type coercion is not allowed.</exception>
+        private static object CoerceType(object existingValue, object newValue)
+        {
+            if (!existingValue.GetType().IsAssignableFrom(newValue.GetType()))
```

Comment:
What scenarios outside of type equality does this check cover?

Reply:
One example would be T vs Nullable<T>

Reply:
Do we support nullable value types?

Reply:
For the model, yes. `DateTimeOffset?` and `DateTime?`. The main scenario here would be that we fetch an entity and select a subset of properties, leaving some of the default system properties empty in the model.

Reply:
In addition, setting null for an entity update/replace call, effectively removing that property.